ui/src/apps/diagnose/components/DiagnoseGenerator.tsx

```diff
@@ -52,18 +64,49 @@ export default function DiagnoseGenerator() {
         >
           <DatePicker showTime />
         </Form.Item>
-        <Form.Item
-          name="rangeDuration"
-          rules={[{ required: true }]}
-          label={t('diagnose.generate.range_duration')}
-        >
-          <Select style={{ width: 120 }}>
-            <Select.Option value={5}>5 min</Select.Option>
-            <Select.Option value={10}>10 min</Select.Option>
-            <Select.Option value={30}>30 min</Select.Option>
-            <Select.Option value={60}>1 hour</Select.Option>
-            <Select.Option value={24 * 60}>1 day</Select.Option>
-          </Select>
+        <Form.Item label={t('diagnose.generate.range_duration')}>
+          <Input.Group compact>
+            <Form.Item
+              name="rangeDuration"
+              rules={[{ required: true }]}
+              noStyle
+            >
+              <Select style={{ width: 120 }}>
+                <Select.Option value={5}>5 min</Select.Option>
+                <Select.Option value={10}>10 min</Select.Option>
+                <Select.Option value={30}>30 min</Select.Option>
+                <Select.Option value={60}>1 hour</Select.Option>
+                <Select.Option value={24 * 60}>1 day</Select.Option>
+                <Select.Option value={0}>Custom</Select.Option>
```

Comment:
Can we support i18n here?

Reply:
I guess it should be 5min (consider other abbreviations 5km, 5kg, etc)

Reply:
er, make sense.

Reply:
Let me add i18n support soon.